pkg/daemon/controller/controller.go

```diff
@@ -128,8 +130,12 @@ func RunController(ctx context.Context, apiServerURL, saCA, saToken string) erro
 }
 
 // NewServiceController returns a pointer to a ServiceController object
-func NewServiceController(ctx context.Context, client client.Client, mgr manager.Manager, nodeName string) *ServiceController {
-	return &ServiceController{client: client, Manager: mgr, ctx: ctx, nodeName: nodeName,
+func NewServiceController(ctx context.Context, client client.Client, mgr manager.Manager,
+	cmdRunner powershell.CommandRunner, nodeName string) *ServiceController {
+	if cmdRunner == nil {
+		cmdRunner = powershell.NewCommandRunner()
+	}
```

Comment:
This is very puzzling. I don't see why `cmdRunner` needs to be passed in.

Reply:
As an alternative I could introduce a `ControllerOptions` struct to be passed in, this could clean things up a bit.

Reply:
`ControllerOptions` would reduce the amount of arguments to the NewController function allow the caller to specify an alternate CommandRunner, and potentially other inputs such as the manager used.

Reply:
I don't know. I would have to see the implementation as I am not sure what you mean by `ControllerOptions`.